packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx

```diff
@@ -205,6 +205,7 @@ export class GroupedListSection extends BaseComponent<IGroupedListSectionProps,
         ref={this._root}
         className={css(groupedListClassNames && groupedListClassNames.group, this._getDroppingClassName())}
         role="presentation"
+        aria-expanded={group ? !group.isCollapsed : undefined}
```

Comment:
it should be true/false. also, a bit odd being on something role=presentation

Reply:
To further expand on this issue, `aria-expanded` can have three possible values. `true`, `false`, and `(null/nonexistent)`. On an expandable item, it should be either true or false. If the item cannot actually be expanded (not as in its expansion is disabled but it's, say, a leaf node), then aria-expanded should not exist there as either a `true` or `false` value.

```
        aria-expanded={group ? !group.isCollapsed : undefined}
```
makes sense to me? If it is a `group`, then it can have expanded/collapsed states, therefore we use the `!group.isCollapsed` value. If it is not a `group`, it should be a leaf node (is my understanding wrong?), therefore it should be `undefined` so that in the rendered DOM there is no `aria-expanded` at all.

Reply:
But it should not be a `role=presentation`. `role=presentation` is not one that "supports" aria-expanded. You should be thinking of a `tree`, or `treegrid`, for such things...

Reply:
@philipzloh That was my understanding for the `aria-expanded` as well. @cschlechty were you thinking something different?

As for putting `aria-expanded` on `role=presentation`, I just checked the W3C standards and it looks like it might actually be best to place it on the button that performs the expand/collapse action. Here's what they have on [one of their Wiki pages][1]:

> When authors use collapsible content, for example, to hide navigation menus or lists of content, the triggering link or button should indicate to screen reader users whether the collapsable content below is in the expanded or in the collapsed state. The aria-expanded attribute is used for this purpose.

What do you two think?

[1]: https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element

Reply:
yes, please use it on the button. You may want to also add aria-controls to show that the button controls that region expanding

Reply:
So, there's the button, and then there's the thing itself. If you have a `tree` and `treeitem`s, then your `treeitem` should have `aria-expanded` states as appropriate, regardless of a `button` that expand/collapses. Furthermore, the `button` that is in your quote is one that "shows/hides" a pane or a menu, for example... it isn't the twisty on lists! If this code is for the twisty on the list, then it should have `aria-hidden=true` (and `role=presentation` if it is a native `<button>` and has implicit `role=button`) because the row itself should carry the `aria-expanded` markup and should be the expected way to toggle the expanded/collapsed state (with a left/right arrow).

Reply:
The expand/collapse button is native `<button>` and already has `aria-hidden=true`.

Additionally, you mention that the button in the quote "shows/hides" a pane or a menu, but the quote also does mention "to hide [...] lists of content". This grouped list is removed from the DOM structure when collapsed. Is this not the same kind of list?

When expanded, it looks like the `list` portion of the GroupedList is made up of the following DOM structure:

```
<div role="list">
  <div class="ms-List-surface" role="presentation">
    <div class="ms-List-page" role="presentation">
      <div  class="ms-List-cell" role="listitem">
        <div role="presentation"> // was originally adding aria-expanded here
          <div class="ms-GroupHeader">
          <div role="list">
...
```

So, for clarification, when you mention `tree` and `treeitem`, is this something I should add to the DOM structure? Or should I add `aria-expanded` to one of the already existing divs, like the div with `role=listitem`?

Reply:
What do you think of @cschlechty's suggestion, using `aria-controls` inline with `aria-expanded`?

Reply:
If you have nested DOM, `aria-controls` is literally redundant (and should not be there). If you don't have nested DOM, then you want it to control the `group`. Sorry I'm not rendering your HTML to answer my own question 😛 

`list`s and `listitem`s are *not expected to be interactive* and should 100% not have `aria-expanded`, and furthermore are likely to be the wrong role.